roles/rsyslog/tasks/inputs/remote/main.yml

```diff
@@ -0,0 +1,37 @@
+---
+# Deploy configuration files
+- name: Install/Update remote input packages and generate configuration files in /etc/rsyslog.d
+  vars:
+    __rsyslog_packages: "{{ __rsyslog_remote_packages }}"
+    __rsyslog_rules: "{{ __rsyslog_remote_rules }}"
+  include_tasks:
+    file: deploy.yml
+
+- name: Create basics input configuration files in /etc/rsyslog.d
+  vars:
+    __rsyslog_input_name: "{{ remote_item.name }}"
+    __rsyslog_input_type: "{{ remote_item.type }}"
+    __rsyslog_packages: []
+    __rsyslog_rules:
+      - name: "input-remote-{{ remote_item.name }}"
+        type: input
+        weight: "11"
+        sections:
+          - options: |-
+              {% if remote_item.udp_port | d() %}
+              # Log messages from remote hosts over UDP
+              input(name="{{ remote_item.name }}" type="imudp" port="{{ remote_item.udp_port | d(__rsyslog_default_port) }}")
+              {% elif remote_item.tcp_port | d() %}
+              # Log messages from remote hosts over plain TCP
+              input(name="{{ remote_item.name }}" type="imptcp" port="{{ remote_item.tcp_port | d(__rsyslog_default_port) }}")
+              {% else %}
+              # Log messages from remote hosts over TLS
+              input(name="{{ remote_item.name }}" type="imtcp" port="{{ remote_item.tcp_tls_port | d(__rsyslog_default_tls_port) }}")
+              {% endif %}
+              {{ lookup('template', 'input_template.j2') }}
```

Comment:
I suggest putting this into a template file `input_remote_template.j2` or something like that

Reply:
Thanks, @richm. I'm updating the pr.

I was wondering which is preferable - sharing templates and making the number of templates as small as possible vs simplifying the task code. It sounds latter, doesn't it? It's not like an analogy like a template is a library function to be shared so that it's supposed to be generic. :)

Reply:
> I was wondering which is preferable - sharing templates and making the number of templates as small as possible vs simplifying the task code. It sounds latter, doesn't it?

In this case, yes, but like anything, it is a judgement call - if the tasks code is "too complex", it should be moved into a template file, or otherwise refactored - for sure the yaml and ansible linters don't like really long lines, and trying to follow rules of indentation can sometimes reduce readability.  On the other hand, moving too much code to template files can also reduce readability if you have to constantly switch back and forth between the tasks and the template to find out what's going on.

Reply:
looking up a template from another template gives you the best of both, doesn't it?

Reply:
Also, would template inheritance help as an alternative to the template lookup? https://jinja.palletsprojects.com/en/2.9.x/templates/#template-inheritance